### PR TITLE
Abstract Message Service for multiple transport protocols

### PR DESCRIPTION
**Pull request by tplooker**
#### Short description of what this resolves:

This PR abstracts the message transport protocols out to leave room for other transport types.

#### Changes proposed in this pull request:

- Remove HttpClient from DefaultMessageService.
- Create IMessageDispatcher contract.
- Create HttpMessageDispatcher implementation for the http and https schemes.

### CODE REVIEW

**Comment by tmarkovski** [comment]
I think there's further abstraction that can be made here. `Uri` is specific to TCP application protocols, but this doesn't always have to be the case. I can deliver a message by copying it to a physical storage.
Making `Uri` to something like `Endpoint` will allow you more freedom. The `HttpDispatcher` can then work with `HttpEndpoint` that has a property `Uri`. Thoughts?

**Comment by tplooker** [comment]
Cant files on local storage be represented in URI format? https://en.wikipedia.org/wiki/File_URI_scheme If the base syntax of endpoint is not a URI, what syntax would be use to decide what protocol to dispatch the message with?

**Comment by tmarkovski** [comment]
> Cant files on local storage be represented in URI format? https://en.wikipedia.org/wiki/File_URI_scheme If the base syntax of endpoint is not a URI, what syntax would be use to decide what protocol to dispatch the message with?

It would need a resolver or a provider pattern. This is sufficient for now most of the time it'll be HTTP anyway.

File: src/AgentFramework.Core/Contracts/IMessageDispatcher.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        /// </summary>
+        string[] TransportSchemes { get; }
+
+        /// <summary>
+        /// Sends a message using the dispatcher.
+        /// </summary>
+        /// <param name="uri">Uri to dispatch the message to.</param>
+        /// <param name="message">Message context to dispatch.</param>
+        /// <returns>A message context.</returns>
+        Task<MessageContext> SendAsync(Uri uri, MessageContext message);
```
Comment (tmarkovski) [inline src/AgentFramework.Core/Contracts/IMessageDispatcher.cs]: For a class named Dispatcher, the action would probably be `DispatchAsync`

Comment (tplooker) [reply src/AgentFramework.Core/Contracts/IMessageDispatcher.cs]: Agreed I have changed this to DispatchAsync

**Pull request merged**